examples/source/1.components/amp-script/index.html

```diff
@@ -484,5 +484,37 @@ <h2 id="translated-greeting"></h2>
         button.addEventListener('click', randomSlideDirection);
       </script>
     </div>  
+    <!--
+        ## Local storage
+        Your script has access to the Web Storage API. This lets you persist user information between browser sessions.
+        Here, we use `localStorage` to track a username.
+        If you set the username, then reload this page, the username will remain set.
+    -->
+    <div>
+      <amp-script layout="fixed-height" script="local-storage-script" height="110" class="sample">
+        <div id="username-display"></div>
+        <div>
+          <label>Enter new username here:</label>
+          <input id="username-input" type="text">
+        </div>
+        <button id="username-submit">Submit</button>
+      </amp-script>
+      <script id="local-storage-script" type="text/plain" target="amp-script">
+        const submit = document.getElementById('username-submit');
+        const input = document.getElementById('username-input');
+        const display = document.getElementById('username-display');
+
+        const oldUsername = localStorage.getItem('username');
+        display.innerHTML = oldUsername ? `Your username is <b>${oldUsername}</b>.` : 'Username is not set';
+
+        function setUsername() {
+          const newUsername = input.value;
+          localStorage.setItem('username', newUsername);
+          display.innerHTML = `Username changed to <b>${newUsername}</b>.`;
```

Comment:
Bad thing can happen if someone puts illegal HTML as the input value, e.g. `<script>hello</script>`:

![image](https://user-images.githubusercontent.com/134745/94472956-652fb880-0180-11eb-80e7-3c2f04e4b78d.png)

Instead of using `innerHTML`, probably better to use `textContent`.

Reply:
This is true. But I'm not sure I want to discourage `innerHTML` for that reason... well, ok, I was conflicted about using it because it's not supported for arbitrary HTML, just in simple cases like mine.

I added `innerHTML` because I wanted to make the username boldface. Your solution reflects a better separation of concerns, but it's substantially longer - and we try to keep these samples short. (Sebastian would probably already say mine is too long.)

Reply:
Ok! I've changed this sample to use `textContent` instead. Could you take another look?